src/subscript/ri_wellmod/ri_wellmod.py

```diff
@@ -0,0 +1,586 @@
+    :param string_list: List of strings to check for match
+
+    :return: List of unique strings string_list matching any element in pattern_list
+    """
+    all_matching = []
+    for pattern in pattern_list:
+        for cur_string in string_list:
+            if fnmatch.fnmatchcase(cur_string, pattern):
+                all_matching.append(cur_string)
+
+    # Ensure unique elements
+    matching_string_set = set(all_matching)
+
+    return list(matching_string_set)
+
+
+def is_init_case(ecl_case):
+    """
+    Check if input Eclipse case name corresponds to an initialized Eclipse run.
+
+    :param ecl_case: Run name (or path) to check (file name with suffix ok)
+
+    :return: True/False
+    """
+    ecl_path = Path(ecl_case)
+    has_grid = (
+        ecl_path.with_suffix(".EGRID").exists()
+        or ecl_path.with_suffix(".GRID").exists()
+    )
+    has_init = ecl_path.with_suffix(".INIT").exists()
+    return has_grid and has_init
+
+
+def has_restart_file(ecl_case):
+    """
+    Check if ecl_case has a restart file
+    (Currently required for LGR creation, to be fixed in next ResInsight release )
+    """
+    ecl_path = Path(ecl_case)
+    return (
+        ecl_path.with_suffix(".UNRST").exists()
+        or ecl_path.with_suffix(".X0000").exists()
+    )
+
+
+def rsp_extract_export_names(well_project, well_path_names):
+    """
+    Extract export well names from ResInsight project
+
+    :param well_project: ResInsight project (rsp) file
+    :param well_path_names: List of well names (as returned by the RI instance)
+
+    :return: Dict export_name[well_path_name] = export_name
+    """
+    doc = xml.dom.minidom.parse(well_project)
+    name_nodes = doc.getElementsByTagName("WellNameForExport")
+    export_names = [node.childNodes[0].nodeValue for node in name_nodes]
+    if len(export_names) != len(well_path_names):
+        logger.error("Could not find export names for all wells - returning empty dict")
+        return {}
+
+    return dict(zip(export_names, well_path_names))
+
+
+def decode_lgr_spec(spec):
+    """
+    Decode LGR spec and return as parsed tuple. For ERT FM replace ',' by ';'.
+
+    :param spec: LGR spec in the format 'wname:ref_i,ref_j,ref_k'
+
+    :return: Tuple (str, int, int, int) if well-formed, None if not
+    """
+    tokens = re.split(",|:|;", spec.strip())
+    if len(tokens) != 4:
+        logger.warning("Malformed LGR spec string: %s", spec)
+        return None
+    wname = tokens[0]
+    try:
+        ref_i = int(tokens[1])
+        ref_j = int(tokens[2])
+        ref_k = int(tokens[3])
+    except ValueError:
+        logger.warning("Unable to integer valued refinements from LGR spec %s", spec)
+        return None
+
+    return (wname, ref_i, ref_j, ref_k)
+
+
+def split_arg_string(arg_string):
+    """
+    Split an arg string of formats 'str1,str2,str3' or 'str1|str2|str3'
+    """
+    return [tok.strip() for tok in re.split(r",|;", arg_string.strip())]
+
+
+def main():
+    """
```

Comment:
Might be hard, but the length and complexity of this function warrants some split-and-refactor, illustrated by the pylint scores: 

ri_wellmod.py:288:0: R0914: Too many local variables (59/15) (too-many-locals)
ri_wellmod.py:288:0: R0912: Too many branches (33/12) (too-many-branches)
ri_wellmod.py:288:0: R0915: Too many statements (143/50) (too-many-statements)

Reply:
Noted, but would rather have this as an issue post-merge :)